app/views/workflows/show.json.jbuilder

```diff
@@ -10,7 +10,9 @@ json.folder_contents (@workflow.folder_contents) do |content|
   json.set! 'type', Pathname(content).file? ? 'file' : 'dir'
   json.set! 'fsurl', Filesystem.new.fs(content)
   json.set! 'fs_base', Filesystem.new.fs(File.dirname(content))
-  json.set! 'terminal_base', OodAppkit.shell.url(path: File.dirname(content)).to_s
+  json.set! 'terminal_base', OODClusters[@workflow.batch_host] && OODClusters[@workflow.batch_host].login_allow? ?
+      OodAppkit.shell.url(path: File.dirname(content), host: @workflow.batch_host).to_s :
```

Comment:
Ok, to clarify that's going to make URL's look like

`.../pun/sys/shell/ssh/owens.osc.edu/users/...`

instead of

`.../pun/sys/shell/ssh/owens/users/...`

Is there ever a case where `login_allow?` would return true and `login.host` would be `nil`?